### PR TITLE
Add ability to assign users in test queue

### PR DESCRIPTION
**Pull request by spectranaut**
Hey Erika, this PR does three things:
1. Makes users key'd by ID. This caused me to changes some of the user setting logic, because it was all based on username before. Also, it led me to find and fix that bug where sometimes all the ATs disappear on the user setting page! :)
2. Adds ability to assign and unassign yourself from runs in the user queue

### CODE REVIEW

**Comment by rwaldron** [comment]
@spectranaut is the reason that you're switching from "array of users" to "object of users by userId key" solely because it may be a faster lookup?

File: client/reducers/users.js [hunk 1]
```diff
@@ -1,50 +1,47 @@
...
 export default (state = initialState, action) => {
     switch (action.type) {
         case GET_USERS: {
             const users = action.payload;
-            return Object.assign({}, state, {
-                users
+            const usersById = {};
+            users.map(u => {
+                usersById[u.id] = u;
             });
```
Comment (rwaldron) [inline client/reducers/users.js:14]:
```js
const usersById = users.reduce((accum, user) => {
  accum[u.id] = u;
  return accum;
}, {});
```

You're not mapping an array's values to a new array, you're reducing an array to object.

**Comment by spectranaut** [comment]
> @spectranaut is the reason that you're switching from "array of users" to "object of users by userId key" solely because it may be a faster lookup?

That's right -- and it makes the code cleaner when you are getting user data.

File: client/components/TestQueueRun/index.jsx [hunk 2]
```diff
@@ -0,0 +1,83 @@
+import React, { Component } from 'react';
```
Comment (evmiguel) [inline client/components/TestQueueRun/index.jsx:1]: This component is super clean and well written!

**Pull request merged**